Sources/SwiftFormatRules/NoParensAroundConditions.swift

```diff
@@ -50,8 +50,12 @@ public final class NoParensAroundConditions: SyntaxFormatRule {
 
   public override func visit(_ node: IfStmtSyntax) -> StmtSyntax {
     let conditions = visit(node.conditions).as(ConditionElementListSyntax.self)!
-    let result = node.withIfKeyword(node.ifKeyword.withOneTrailingSpace())
+    var result = node.withIfKeyword(node.ifKeyword.withOneTrailingSpace())
       .withConditions(conditions)
+      .withBody(CodeBlockSyntax(visit(node.body)))
+    if let elseBody = node.elseBody {
+      result = result.withElseBody(visit(elseBody))
+    }
     return StmtSyntax(result)
   }
 
```

Comment:
This is the closest GitHub will let me leave a comment to something that was collapsed...

It looks like we don't recurse into `tup` in `visit(_: ConditionElementSyntax)` starting on line 62, so we might still be missing some (admittedly disgusting) cases like the inner parentheses in this one:

```swift
if foo.someCall({ if (x) {} }) {}
```

Can you test that too?

Reply:
That's a good point. This also made me realize that we need to visit the child nodes in the early exit path for many other node types here. I fixed those as well, and added relevant tests.